arroyo/processing/strategies/run_task_in_threads.py

```diff
@@ -53,6 +54,7 @@ def __init__(
         self.__next_step = next_step
         self.__closed = False
         self.__invalid_messages = InvalidMessageState()
+        self.__abandon_messages_on_shutdown = abandon_messsages_on_shutdown
```

Comment:
Technically join may not only be executed at shutdown. 
I would rename it as abandon messages on join.